app/controllers/users_controller.rb

```diff
@@ -1740,6 +1740,68 @@ def bookmarks
     end
   end
 
+  USER_MENU_BOOKMARKS_LIST_LIMIT = 20
+  def user_menu_bookmarks
+    user = fetch_user_from_params
+    guardian.ensure_can_see_notifications!(user)
+    user_guardian = Guardian.new(user)
+
+    if params[:limit]
+      limit = params[:limit].to_i
+      limit = limit.clamp(1..USER_MENU_BOOKMARKS_LIST_LIMIT)
+    else
+      limit = USER_MENU_BOOKMARKS_LIST_LIMIT
+    end
+
+    reminder_notifications = user
+      .notifications
+      .visible
+      .includes(:topic)
+      .where(read: false, notification_type: Notification.types[:bookmark_reminder])
+      .limit(limit)
+      .to_a
+
+    if reminder_notifications.size < limit
+      exclude_bookmark_ids = reminder_notifications
+        .filter_map { |notification| notification.data_hash[:bookmark_id] }
+
+      bookmark_list = UserBookmarkList.new(
+        user: user,
+        guardian: user_guardian,
+        params: {
+          per_page: limit - reminder_notifications.size
```

Comment:
Why did we choose to use `per_page` here instead of `limit`?

Reply:
I'm not sure I'm understanding your question, but this is what the `UserBookmarkList` class expects:

https://github.com/discourse/discourse/blob/41197edb515ab637c1c4e36a0dc0820bafd52b82/app/models/user_bookmark_list.rb#L17